compiler/ilgen/OMRIlBuilder.hpp

```diff
@@ -255,6 +255,18 @@ class IlBuilder : public TR::IlInjector
    TR::IlValue *ConvertTo(TR::IlType *t, TR::IlValue *v);
    TR::IlValue *UnsignedConvertTo(TR::IlType *t, TR::IlValue *v);
 
+   /**
+    * @brief Coerce an IlValue to a given type
+    * @param type is the target type of the coercion
+    * @param value is the value to be coerced
+    * @return the IlValue coerced to the specified type
+    *
+    * This service acts as a kind of `reinterpret_cast` for JitBuilder IlValues.
+    * Given some input value, it produces a new value that represents the same
+    * bit pattern as the original one, but with a different type.
+    */
+   TR::IlValue* CoerceTo(TR::IlType* type, TR::IlValue* value);
```

Comment:
Maybe we should borrow the verb from C++ here and call it `ReinterpretAs` or `ReinterpretCast` ?

The primary hit on google for "coerce a type" yields examples that are more like our ConvertTo service.

Reply:
Agree, coerce to me usually means "different representation, equivalent value".  What about
What about some derivative of "retype bits" or "force type to" ?

Reply:
Actually, as I explained in the issue, I specifically wanted to avoid borrowing C++'s `reinterpret_cast` because it specifically behaves differently (see https://en.cppreference.com/w/cpp/language/reinterpret_cast and https://godbolt.org/z/jZI_7q).

Reply:
Not sure the pointer issue you mentioned is legit--reinterpret seems to be a straightforward (if restrictive) type-changing operation. FWIW, llvm calls this op [bitcast to](https://llvm.org/docs/LangRef.html#bitcast-to-instruction), and there's a new [`bit_cast`](https://en.cppreference.com/w/cpp/numeric/bit_cast) in c++20.

Reply:
Perhaps I didn't explain my concern well enough. While it's true that the type changing aspect of`reinterpret_cast` is straightforward, the restrictions on it make use-cases for it are very different than the use-cases for the existing "CoerceTo" service.

I'd be okay with renaming it to `BitCastTo()` to maybe `ConvertBitsTo(). @mstoodle thoughts?

Reply:
I'd go with `ConvertBitsTo()` for synergy with `ConvertTo()`. Any opinions on whether we should create a new `ConvertValueTo()` and deprecate `ConvertTo()` at some later date?

Reply:
:+1: for `ConvertValueTo()`

Reply:
Fixed! Settled on `ConvertBitsTo()`.